site/content/includes/installation/nginx-plus/nginx-plus-secret.md

```diff
@@ -0,0 +1,13 @@
+---
+docs: "DOCS-000"
+---
+
+Place the JWT in a file called `license.jwt`. Create a Kubernetes Secret using the contents of the JWT file.
+
+```shell
+kubectl create secret generic nplus-license --from-file license.jwt -n nginx-gateway
+```
+
+You can now delete the `license.jwt` file.
+
+If you need to update the JWT at any time, update the `license.jwt` field in the Secret using `kubectl edit` and apply the changes. The volume mount containing the Secret will be updated by the kubelet when it syncs the Pod.
```

Comment:
What does it mean to "sync the Pod"? When does that happen? I thought updating a secret mounted to a Pod would require a Pod restart for the changes to the secret to be picked up.

Reply:
I don't know the default timing off hand, but in my testing it was within a minute. https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

No need to restart the pod (doesn't work for subPath mounts)

Reply:
Gotcha. Does nginx pick up the rotated secret without a reload?

Reply:
No, looks like it requires a reload in order to pick up the new license file.

So that's a problem...I originally had the code to write that file and reload instead of relying on the volume mount, but I think I'll have to add all of that back in...

Reply:
bummer 🙁

Reply:
Added logic to track these secrets and write them to nginx. It's probably not the cleanest solution and has room for improvement, but due to the time crunch I just wanted something that works.